multicluster/controllers/multicluster/label_identity_export_controller.go

```diff
@@ -0,0 +1,296 @@
+	clusterID := resExport.Spec.ClusterID
+	clusterLabelSet := sets.String{}
+	for _, l := range resExport.Spec.LabelIdentities.NormalizedLabels {
+		clusterLabelSet.Insert(l)
+	}
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	originalClusterLabels, ok := r.clusterToLabels[clusterID]
+	if ok {
+		addedLabels := clusterLabelSet.Difference(originalClusterLabels)
+		deletedLabels := originalClusterLabels.Difference(clusterLabelSet)
+		return r.refreshLabelIdentityImports(ctx, resExport, addedLabels, deletedLabels, clusterID)
+	}
+	return r.refreshLabelIdentityImports(ctx, resExport, clusterLabelSet, nil, clusterID)
+}
+
+// refreshLabelIdentityImports computes the new LabelIdentityImport objects to be created, and old
+// LabelIdentityImport objects to be deleted.
+func (r *LabelIdentityExportReconciler) refreshLabelIdentityImports(ctx context.Context,
+	resExport *mcsv1alpha1.ResourceExport, addedClusterLabels, deletedClusterLabels sets.String, clusterID string) error {
+	var addedLabels []string
+	var deletedLabels []string
+	for l := range addedClusterLabels {
+		if _, ok := r.labelsToClusters[l]; !ok {
+			// This is a new label identity in the entire ClusterSet.
+			addedLabels = append(addedLabels, l)
+		}
+	}
+	for l := range deletedClusterLabels {
+		deletedCluster := sets.NewString(clusterID)
+		if clusters, ok := r.labelsToClusters[l]; ok && clusters.Equal(deletedCluster) {
+			// The cluster where the label identity is being deleted was the only cluster that has
+			// the label identity. Hence, the label identity is no longer present in the ClusterSet.
+			deletedLabels = append(deletedLabels, l)
+		}
+	}
+	if len(addedLabels)+len(deletedLabels) == 0 {
+		return nil
+	}
+	reconcileFailedLabels := sets.NewString()
+	r.handleLabelIdentitiesDelete(ctx, deletedLabels, reconcileFailedLabels, clusterID, resExport)
+	r.handleLabelIdentitiesAdd(ctx, addedLabels, reconcileFailedLabels, clusterID, resExport)
+	if len(reconcileFailedLabels) > 0 {
+		return fmt.Errorf("failed to reconcile LabelIdentityImport for labels %v", reconcileFailedLabels)
+	}
+	return nil
+}
+
+// handleLabelIdentitiesDelete deletes LabelIdentityImports of label identities that no longer exists
+// in the ClusterSet. Note that the ID of a label identity is only released if the deletion of its
+// LabelIdentityImport succeeded.
+func (r *LabelIdentityExportReconciler) handleLabelIdentitiesDelete(ctx context.Context,
+	deletedLabels []string, reconcileFailedLabels sets.String, clusterID string, resExport *mcsv1alpha1.ResourceExport) {
+	for _, label := range deletedLabels {
+		existingLabelImport := &mcsv1alpha1.LabelIdentityImport{}
+		resNamespaced := types.NamespacedName{
+			Name:      hashLabelIdentity(label),
+			Namespace: resExport.Namespace,
+		}
+		err := r.Client.Get(ctx, resNamespaced, existingLabelImport)
+		if err != nil && !apierrors.IsNotFound(err) {
+			klog.ErrorS(err, "failed to get LabelIdentityImport for stale label", "label", label)
```

Comment:
```suggestion
			klog.ErrorS(err, "Failed to get LabelIdentityImport for stale label", "label", label)
```

Reply:
Looks like you continue to delete it when it's not found error. I feel it's unnecessary to check if it's existing or not since your purpose is to clean up it, better call delete method and ignore not found error.

Reply:
Again this is related to https://github.com/antrea-io/antrea/pull/3664#discussion_r898054760, should fix it together with other resource deletions